Mathlib/Combinatorics/SimpleGraph/VertexCover.lean

```diff
@@ -0,0 +1,192 @@
+/-
+Copyright (c) 2025 Vlad Tsyrklevich. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vlad Tsyrklevich
+-/
+import Mathlib.Combinatorics.SimpleGraph.Clique
+import Mathlib.Data.ENat.Lattice
+import Mathlib.Data.Set.Card
+
+/-!
+# Vertex cover
+
+A *vertex cover* of a simple graph is a set of vertices such that every edge is incident to at least
+one of the vertices in the set.
+
+## Main definitions
+
+* `SimpleGraph.IsVertexCover G C`: Predicate that `C` is a vertex cover of `G`.
+* `SimpleGraph.minVertexCover G`: The size of the minimal vertex cover for `G`.
+-/
+
+namespace SimpleGraph
+
+variable {V W : Type*} {G : SimpleGraph V} {H : SimpleGraph W}
+
+section IsVertexCover
+
+/-- `C` is a vertex cover of `G` if every edge in `G` is incident to at least one vertex in `C`. -/
+def IsVertexCover (G : SimpleGraph V) (c : Set V) : Prop :=
+  ∀ v w : V, G.Adj v w → v ∈ c ∨ w ∈ c
+
+theorem isVertexCover_iff {c : Set V} :
+  IsVertexCover G c ↔ ∀ v w : V, G.Adj v w → v ∈ c ∨ w ∈ c := Iff.rfl
+
+@[simp]
+theorem isVertexCover_empty_iff_bot : IsVertexCover G ∅ ↔ G = ⊥ := by
+  constructor
+  · intro h
+    contrapose! h
+    simp [IsVertexCover, ne_bot_iff_exists_adj.mp h]
+  · simp_all [IsVertexCover]
+
+theorem not_isVertexCover_of_ne_bot (h : G ≠ ⊥) : ¬IsVertexCover G ∅ :=
+  iff_false_right h |>.mp isVertexCover_empty_iff_bot
+
+@[simp]
+theorem isVertexCover_univ : IsVertexCover G Set.univ := by
+  simp [IsVertexCover]
+
+@[simp]
+theorem isVertexCover_bot (c : Set V) : IsVertexCover ⊥ c := by
+  simp [IsVertexCover]
+
+theorem IsVertexCover.mono (c d : Set V) (hcd : c ⊆ d) (hc : IsVertexCover G c) :
+    IsVertexCover G d := by
+  grind [IsVertexCover]
+
+/-- A set `c` is a vertex cover iff the complement of `c` is an independent set. -/
+theorem isVertexCover_iff_isIndepSet_compl {c : Set V} : IsVertexCover G c ↔ G.IsIndepSet cᶜ := by
+  constructor
+  · grind [IsVertexCover, Set.Pairwise]
+  · intro hi v w hadj
+    by_cases hh : v ∈ c ∨ w ∈ c
+    · grind
+    · rw [not_or] at hh
+      exact False.elim <| hi hh.1 hh.2 (Adj.ne hadj) hadj
+
+theorem IsVertexCover.preimage_of_relHom (f : G →g H) {c : Set W} (hc : IsVertexCover H c) :
```

Comment:
The fact that simple graph homs are implemented using `RelHom` is an implementation detail. Furthermore, this is not an hypothesis, so shouldn't use `_of`. Finally, I think it's unnecessary here
```suggestion
theorem IsVertexCover.preimage (f : G →g H) {c : Set W} (hc : IsVertexCover H c) :
```